src/openrct2/peep/Peep.cpp

```diff
@@ -3077,7 +3071,7 @@ void Peep::PerformNextAction(uint8_t& pathing_result, TileElement*& tile_result)
 
     auto newLoc = *loc;
     CoordsXY truncatedNewLoc = newLoc.ToTileStart();
-    if (truncatedNewLoc.x == next_x && truncatedNewLoc.y == next_y)
+    if (truncatedNewLoc.x == NextLoc.x && truncatedNewLoc.y == NextLoc.y)
```

Comment:
Can't this be just: 
```suggestion
    if (truncatedNewLoc == NextLoc)
```
Or 
```suggestion
    if (truncatedNewLoc == CoordsXY(NextLoc))
```

Reply:
There is no comparison operator defined for CoordsXY only for tile cords.

Reply:
:O
I think one would be very useful.

Reply:
Yeah I will probably make one. I see CoordsXYZ has one so just looks like it was missed.